packages/in_app_purchase/lib/src/in_app_purchase_connection/app_store_connection.dart

```diff
@@ -49,3 +106,61 @@ class AppStoreConnection implements InAppPurchaseConnection {
     return productDetailsResponse;
   }
 }
+
+class _TransactionObserver implements SKTransactionObserverWrapper {
+  Completer<List<SKPaymentTransactionWrapper>> _restoreCompleter;
+  List<SKPaymentTransactionWrapper> _restoredTransactions;
+
+  Future<List<SKPaymentTransactionWrapper>> getRestoredTransactions(
+      {@required SKPaymentQueueWrapper queue, String applicationUserName}) {
+    assert(queue != null);
+    _restoreCompleter = Completer();
+    queue.restoreTransactions(applicationUserName: applicationUserName);
+    return _restoreCompleter.future;
+  }
+
+  void cleanUpRestoredTransactions() {
+    _restoredTransactions = null;
+    _restoreCompleter = null;
+  }
+
+  /// Triggered when any transactions are updated.
+  void updatedTransactions({List<SKPaymentTransactionWrapper> transactions}) {
+    assert(_restoreCompleter != null);
```

Comment:
What happens if we call `cleanUpRestoredTransactions()` and then this triggers from a new purchase?